wallet/src/hooks/User/index.jsx

```diff
@@ -105,58 +110,55 @@ export const UserProvider = ({ children }) => {
 
     // Listen for messages
     messageEventHandler = async function (event) {
-      console.log('Message from server ', JSON.parse(event.data));
-      const parsed = JSON.parse(event.data);
+      const parsedEvent = JSON.parse(event.data);
+      console.log('Message from server ', parsedEvent);
+
       const { nullifierKey, zkpPrivateKey } = await retrieveAndDecrypt(compressedZkpPublicKey);
-      if (parsed.type === 'sync') {
-        await parsed.historicalData
+
+      if (parsedEvent.type === 'sync') {
+        await parsedEvent.historicalData
           .sort((a, b) => a.block.blockNumberL2 - b.block.blockNumberL2)
           .reduce(async (acc, curr) => {
             await acc; // Acc is a promise so we await it before processing the next one;
             return blockProposedEventHandler(curr, [zkpPrivateKey], [nullifierKey]); // TODO Should be array
           }, Promise.resolve());
 
-        // We want to verify that the received block is from the current contract deployment.
-        // for this, we store the lastBlock received in lastBlock, and compare the hash with the previousBlockHash
-        // from the received block. If they don't match, then there has been a redeployment.
-        if (Number(parsed.maxBlock) !== 1) {
-          if (
-            parsed.historicalData[parsed.historicalData.length - 1].block.previousBlockHash !==
-              lastBlock.blockHash &&
-            Number(lastBlock.blockHash) !== 0
-          ) {
-            // resync
-            console.log('Resync DB');
-            emptyStoreBlocks();
-            emptyStoreTimber();
-            Storage.shieldAddressSet();
-          } else if (
-            parsed.historicalData[parsed.historicalData.length - 1].block.previousBlockHash ===
-              lastBlock.blockHash ||
-            Number(lastBlock.blockHash) === 0
-          ) {
-            setLastBlock(parsed.historicalData[parsed.historicalData.length - 1].block);
```

Comment:
You removed the instruction `setLastBlock`, but I didn't see it anywhere in the changes you did. Is this correct? Won't this cause any undesired colateral effect?

Reply:
Good point, let me check.